### PR TITLE
Add tests + examples + bump version

### PR DESCRIPTION
**Pull request by remithomas**
I did some improvements:

- Add tests 
- Add some examples
- Bump version number
- Upgrade `bs-platform` to `3.1.4`

@Ahrefs which CI do you prefer to add (circleCI, TravisCI, other...) ?

### CODE REVIEW

File: src/dotenv.re [hunk 1]
```diff
@@ -30,10 +30,10 @@ type config = {
   encoding: option(string),
 };
 
-let config = (~cfg=?, ()) =>
+let config = (~cfg=?, ()) =>
```
Comment (Khady) [inline src/dotenv.re]: is it the new refmt that adds trailing whitespaces?

File: examples/complex.bs.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+// Generated by BUCKLESCRIPT VERSION 3.1.5, PLEASE EDIT WITH CARE
```
Comment (Khady) [inline examples/complex.bs.js]: I don't think we need to commit javascript code of examples.

File: __tests__/BsDotenv_test.bs.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+// Generated by BUCKLESCRIPT VERSION 3.1.5, PLEASE EDIT WITH CARE
```
Comment (Khady) [inline __tests__/BsDotenv_test.bs.js]: Do we need to commit those files to run the test? I don't think so, but better to check.

File: .gitignore [hunk 4]
```diff
@@ -6,3 +6,10 @@ node_modules
 npm-debug.log*
 yarn-debug.log*
 yarn-error.log*
+
+# tests
+coverage
+.nyc_output
+
+# Build
+src/*.bs.js
```
Comment (Khady) [inline .gitignore]: maybe just ignore all `*.bs.js` files

**Comment by Khady** [comment]
Thanks. I made a few comments. It looks good in general.

I've asked to enable CI for the project, will let you know once I now if we can use travis or circle.

File: examples/complex.bs.js (hunk 2 above)
Comment (remithomas) [reply examples/complex.bs.js]: You right, I remove them.

File: __tests__/BsDotenv_test.bs.js (hunk 3 above)
Comment (remithomas) [reply __tests__/BsDotenv_test.bs.js]: I added a pre hook to build the file. Initially I was thinking to do in the CI config file but good idea !

**Pull request merged**